### PR TITLE
[13.0][FIX] sale_coupon_multi_use: Replace sudo by with_user

### PR DESCRIPTION
**Pull request by jcoux**
Replace sudo by with_user.

### CODE REVIEW

File: sale_coupon_multi_use/models/sale_order_line.py [hunk 1]
```diff
@@ -19,6 +19,6 @@ def unlink(self):
         consumption_lines = self.mapped("coupon_consumption_line_ids")
         if consumption_lines:
             # Remove related consumption lines
-            # Using sudo because no user can remove consumption lines
-            consumption_lines.sudo().unlink()
+            # Using super user because no user can remove consumption lines
+            consumption_lines.with_user(SUPERUSER_ID).unlink()
```
Comment (rousseldenis) [inline sale_coupon_multi_use/models/sale_order_line.py:23]: Actually, sudo() still exists and apply automatically SUPERUSER_ID. But with_user() allows to 'do the operations with high privileges keeping the user details - logging for instance'.

sudo : https://github.com/odoo/odoo/blob/13.0/odoo/models.py#L5150

with_user: https://github.com/odoo/odoo/blob/13.0/odoo/models.py#L5161


So, you can let sudo() here :smiley:

Comment (jcoux) [reply sale_coupon_multi_use/models/sale_order_line.py:23]: Now it's done: it will be more easy to migrate in future version ;)

**Review by ValentinVinagre — APPROVED** [review]
LGTM 👍

**Comment by jcoux** [comment]
Seems this PR is useless: close it.

**Pull request closed without merging**